controllers/space/space_controller.go

```diff
@@ -240,6 +241,21 @@ func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, space *toolchainv1a
 		if err := r.Client.Update(context.TODO(), space); err != nil {
 			return norequeue, r.setStatusProvisioningFailed(logger, space, err)
 		}
+
+		// update provisioned namespace list
+		oldProvisionedNamespaces := space.Status.ProvisionedNamespaces
+		requeue, err := r.setStatusProvisionedNamespaces(space, nsTmplSet.Status.ProvisionedNamespaces)
+		if err != nil {
+			err = errs.Wrap(err, "error setting provisioned namespaces")
+			// restore old list of provisioned namespace since update of the Status.ProvisionedNamespace field failed.
+			space.Status.ProvisionedNamespaces = oldProvisionedNamespaces
+			return norequeue, r.setStatusProvisioningFailed(logger, space, err)
```

Comment:
I think if the status update is failed that we can't do anything more beyond logging the error and returning an error so a new reconcile is queued. Updating status to report an error with updating the status is problematic :)
There is also no need to restore the status in the object you have in the memory right before exiting the reconcile. This object is about to get garbage collected :)
```suggestion
			logger.Error(err, "failed to update space status")
			return norequeue, err
```